core/src/main/java/org/jdbi/v3/core/generic/GenericTypes.java

```diff
@@ -116,9 +113,20 @@ private GenericTypes() {
      * @return the resolved type.
      */
     public static Type resolveType(Type type, Type contextType) {
-        return TypeToken.of(contextType)
-                .resolveType(type)
-                .getType();
+        return GenericTypeReflector.resolveType(type, contextType);
+    }
+
+    /**
+     * Resolves the <code>type</code> parameter in the context of <code>contextType</code>. For example, if
+     * <code>type</code> is <code>List.class.getMethod("get", int.class).getGenericReturnType()</code>, and
+     * <code>contextType</code> is <code>List&lt;String&gt;</code>, this method returns <code>String.class</code>
+     * @param type the type to be resolved in the scope of <code>contextType</code>
+     * @param contextType the context type in which <code>type</code> is interpreted to resolve the type.
+     * @return the resolved type.
+     */
+    public static Type resolveType(TypeVariable<? extends Class<?>> type, Type contextType) {
```

Comment:
I'm unclear on what this new method is doing. How do the semantics of this method vary from the existing one?

I'm also a little concerned that this new method overload takes over 7 of the 15 existing references to the existing `resolveType` method. Is this intentional?

Would you be open to renaming this method, e.g. to `resolveTypeVariable()` to prevent ambiguity?

Reply:
For reference, the `resolveType(TypeVariable, Type)` overload shouldn't be needed. The usual `resolveType(Type, Type)` covers it. There is probably a small difference in performance, but shouldn't be anything relevant.

Reply:
It came in one of the intermediate steps, before the new geantyref methods existed.  I'll remove it.